devito/types/dense.py

```diff
@@ -111,8 +111,14 @@ def _allocate_memory(func):
         def wrapper(self):
             if self._data is None:
                 debug("Allocating memory for %s%s" % (self.name, self.shape_allocated))
+                try:
+                    distributor = self._grid._distributor
+                except AttributeError:
+                    # Data not attached to a grid
```

Comment:
I don't understand this try-catch? what case is it covering?

Reply:
Can't remember why it was there originally but is not needed and has been removed.